updater/fetchers/nodejs/nodejs.go

```diff
@@ -0,0 +1,195 @@
+
+type nodejsAdvisory struct {
+	ID                 int      `json:"id"`
+	CreatedAt          string   `json:"created_at"`
+	UpdatedAt          string   `json:"updated_at"`
+	PublishDate        string   `json:"publish_date"`
+	Title              string   `json:"title"`
+	Author             string   `json:"author"`
+	ModuleName         string   `json:"module_name"`
+	CVES               []string `json:"cves"`
+	VulnerableVersions string   `json:"vulnerable_versions"`
+	PatchedVersions    string   `json:"patched_versions"`
+	Slug               string   `json:"slug"`
+	Overview           string   `json:"overview"`
+	Recommandation     string   `json:"recommandation"`
+	References         string   `json:"references"`
+	LegacySlug         string   `json:"legacy_slug"`
+	AllowedScopes      []string `json:"allowed_scopes"`
+	CvesVector         string   `json:"cves_vector"`
+	CvssScore          float32  `json:"cvss_score"`
+}
+
+type nodejsAdvisories struct {
+	Total   int              `json:"total"`
+	Count   int              `json:"count"`
+	Offset  int              `json:"offset"`
+	Results []nodejsAdvisory `json:"results"`
+}
+
+// NodejsFetcher implements updater.Fetcher for the Node Security Project
+// (https://nodesecurity.io).
+type NodejsFetcher struct{}
+
+func init() {
+	updater.RegisterFetcher("nodejs", &NodejsFetcher{})
+}
+
+// FetchUpdate fetches vulnerability updates from the Node Security Project.
+func (fetcher *NodejsFetcher) FetchUpdate(datastore database.Datastore) (resp updater.FetcherResponse, err error) {
+	log.Info("fetching Nodejs vulnerabilities")
+
+	// Download JSON.
+	r, err := http.Get(url)
+	if err != nil {
+		log.Errorf("could not download Nodejs's update: %s", err)
+		return resp, cerrors.ErrCouldNotDownload
+	}
+
+	defer r.Body.Close()
+
+	// Get the latest date of the latest update's JSON data
+	latestUpdate, err := datastore.GetKeyValue(updaterFlag)
+
+	if err != nil {
+		return resp, err
+	}
+
+	// Unmarshal JSON.
+	var advisories nodejsAdvisories
+	err = json.NewDecoder(r.Body).Decode(&advisories)
+	if err != nil {
+		log.Errorf("could not unmarshal Nodejs's JSON: %s", err)
+		return resp, cerrors.ErrCouldNotParse
+	}
+
+	resp.Vulnerabilities, resp.FlagValue = parseNodejsAdvisories(advisories.Results, latestUpdate)
+	resp.FlagName = updaterFlag
+
+	return resp, nil
+}
+
+func parseNodejsAdvisories(advisories []nodejsAdvisory, latestUpdate string) (vulnerabilities []database.Vulnerability, newUpdated string) {
+	mvulnerabilities := make(map[string]*database.Vulnerability)
+
+	for _, advisory := range advisories {
+		for _, vulnName := range advisory.CVES {
+			if latestUpdate >= advisory.UpdatedAt {
+				break
+			}
+			if advisory.UpdatedAt > newUpdated {
+				newUpdated = advisory.UpdatedAt
+			}
+			// Get or create the vulnerability.
+			vulnerability, vulnerabilityAlreadyExists := mvulnerabilities[vulnName]
+			if !vulnerabilityAlreadyExists {
```

Comment:
I prefer to keep these separated, the 'vulnerability' is used in the later codes.